### PR TITLE
Fix buffer read and added missing tests

### PR DESCRIPTION
**Pull request by AhmedKorim**
## Overview
- Fixed the buffer passed to `DataView` 
- Adding more unit tests for proposals decoding classes

### CODE REVIEW

File: packages/sdk-core/src/__test__/proposals.spec.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ describe('test various conversion functions', () => {
     assert.equal(anchorCreateDecoded.header.nonce, lastLeafIndex);
     assert.equal(anchorCreateDecoded.encodedCall, encodedCall);
   });
-  it('Should encode and decode an evm proposal', () => {
+  it('Should encode and decode  evm proposal', () => {
```
Comment (dutterbutter) [inline packages/sdk-core/src/__test__/proposals.spec.ts:106]:
```suggestion
  it('Should encode and decode evm proposal', () => {
```

File: packages/sdk-core/src/__test__/proposals.spec.ts [hunk 2]
```diff
@@ -113,4 +113,264 @@ describe('test various conversion functions', () => {
     assert.equal(eVMProposalDecoded.nonce, tx.nonce);
     assert.equal(eVMProposalDecoded.chainId, tx.chainId);
   });
+
+  it('Should encode and decode  refresh vote proposal', () => {
```
Comment (dutterbutter) [inline packages/sdk-core/src/__test__/proposals.spec.ts:117]:
```suggestion
  it('Should encode and decode refresh vote proposal', () => {
```

File: packages/sdk-core/src/__test__/proposals.spec.ts [hunk 3]
```diff
@@ -113,4 +113,264 @@ describe('test various conversion functions', () => {
...
+
+    const refreshVoteProposal = new RefreshVoteProposal(nonce, publicKey);
+    const refreshVoteProposalEncoded = refreshVoteProposal.toU8a();
+    const refreshVoteProposalDecoded = RefreshVoteProposal.fromBytes(refreshVoteProposalEncoded);
+
+    assert.equal(refreshVoteProposalDecoded.nonce, nonce);
+    assert.equal(refreshVoteProposalDecoded.publicKey, publicKey);
+  });
+
+  it('Should encode and decode  proposer set update proposal', () => {
```
Comment (dutterbutter) [inline packages/sdk-core/src/__test__/proposals.spec.ts:129]:
```suggestion
  it('Should encode and decode proposer set update proposal', () => {
```

File: packages/sdk-core/src/__test__/proposals.spec.ts [hunk 4]
```diff
@@ -113,4 +113,264 @@ describe('test various conversion functions', () => {
...
+
+    const tokenAddProposal = new TokenAddProposal(header, newTokenAddress);
+    const tokenAddProposalEncoded = tokenAddProposal.toU8a();
+    const tokenAddProposalDecoded = TokenAddProposal.fromBytes(tokenAddProposalEncoded);
+
+    assert.equal(tokenAddProposalDecoded.newTokenAddress, newTokenAddress);
+    assert.equal(tokenAddProposalDecoded.header.toString(), header.toString());
+  });
+
+  it('Should encode and decode  token remove proposal', () => {
```
Comment (dutterbutter) [inline packages/sdk-core/src/__test__/proposals.spec.ts:172]:
```suggestion
  it('Should encode and decode token remove proposal', () => {
```

File: packages/sdk-core/src/__test__/proposals.spec.ts [hunk 5]
```diff
@@ -113,4 +113,264 @@ describe('test various conversion functions', () => {
...
+
+    const tokenRemoveProposal = new TokenRemoveProposal(header, removedTokenAddress);
+    const tokenRemoveProposalEncoded = tokenRemoveProposal.toU8a();
+    const tokenRemoveProposalDecoded = TokenRemoveProposal.fromBytes(tokenRemoveProposalEncoded);
+
+    assert.equal(tokenRemoveProposalDecoded.removeTokenAddress, removedTokenAddress);
+    assert.equal(tokenRemoveProposalDecoded.header.toString(), header.toString());
+  });
+
+  it('Should encode and decode  wrapping fee update proposal', () => {
```
Comment (dutterbutter) [inline packages/sdk-core/src/__test__/proposals.spec.ts:194]:
```suggestion
  it('Should encode and decode wrapping fee update proposal', () => {
```

File: packages/sdk-core/src/__test__/proposals.spec.ts [hunk 6]
```diff
@@ -113,4 +113,264 @@ describe('test various conversion functions', () => {
...
+
+    const wrappingFeeUpdateProposal = new WrappingFeeUpdateProposal(header, newFee);
+    const wrappingFeeUpdateProposalEncoded = wrappingFeeUpdateProposal.toU8a();
+    const wrappingFeeUpdateProposalDecoded = WrappingFeeUpdateProposal.fromBytes(wrappingFeeUpdateProposalEncoded);
+
+    assert.equal(wrappingFeeUpdateProposalDecoded.newFee, newFee);
+    assert.equal(wrappingFeeUpdateProposalDecoded.header.toString(), header.toString());
+  });
+
+  it('Should encode and decode  min withdraw limit proposal', () => {
```
Comment (dutterbutter) [inline packages/sdk-core/src/__test__/proposals.spec.ts:216]:
```suggestion
  it('Should encode and decode min withdraw limit proposal', () => {
```

File: packages/sdk-core/src/__test__/proposals.spec.ts [hunk 7]
```diff
@@ -113,4 +113,264 @@ describe('test various conversion functions', () => {
...
+
+    const maxDepositLimitProposal = new MaxDepositLimitProposal(header, maxDepositLimitBytes);
+    const maxDepositLimitProposalEncoded = maxDepositLimitProposal.toU8a();
+    const maxDepositLimitProposalDecoded = MaxDepositLimitProposal.fromBytes(maxDepositLimitProposalEncoded);
+
+    assert.equal(maxDepositLimitProposalDecoded.maxDepositLimitBytes, maxDepositLimitBytes);
+    assert.equal(maxDepositLimitProposalDecoded.header.toString(), header.toString());
+  });
+
+  it('Should encode and decode resourceId update  proposal', () => {
```
Comment (dutterbutter) [inline packages/sdk-core/src/__test__/proposals.spec.ts:260]:
```suggestion
  it('Should encode and decode resourceId update proposal', () => {
```

**Review by dutterbutter — CHANGES_REQUESTED** [review]
A lot of variables look like they can be reused rather than redeclared each test. Perhaps we can improve the redundancy there a bit.

**Comment by AhmedKorim** [comment]
@dutterbutter  can you say more?

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**